client/src/unifyfs-stdio.c

```diff
@@ -1559,7 +1573,12 @@ void UNIFYFS_WRAP(rewind)(FILE* stream)
         /* lookup stream */
         unifyfs_stream_t* s = (unifyfs_stream_t*) stream;
 
-        /* TODO: check that stream is active */
+        /* check that stream is active */
+        int fd = s->fd;
+        if (s->fd < 0) {
+            errno = EBADF;
+            return;
+        }
```

Comment:
This should be good for review now. It's currently in three logically separate commits, but can squash them if need be.

One question though. One of the tests I wrote was making `rewind` fail when called on a non-open stream. I noticed this `TODO` within most of our `stdio` wrappers and did the check in this way, which made the test pass. However, not sure if this is the right approach as I'm assuming there is a reason the check hadn't been done in any of the wrappers yet.

Reply:
Noticed I assigned to a local var but didn't use. I'll fix that.

And I verified that `unifyfs_stream_init` does also set `fd` back to -1.
https://github.com/LLNL/UnifyFS/blob/872cb269f23d8e6473ba43032b47e34d5259769b/client/src/unifyfs-stdio.c#L1851-L1852